redash/settings/__init__.py

```diff
@@ -48,9 +49,77 @@ def all_settings():
 SCHEMAS_REFRESH_SCHEDULE = int(os.environ.get("REDASH_SCHEMAS_REFRESH_SCHEDULE", 30))
 
 AUTH_TYPE = os.environ.get("REDASH_AUTH_TYPE", "api_key")
-ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
 INVITATION_TOKEN_MAX_AGE = int(os.environ.get("REDASH_INVITATION_TOKEN_MAX_AGE", 60 * 60 * 24 * 7))
 
+# Whether and how to redirect non-HTTP requests to HTTPS. Disabled by default.
+ENFORCE_HTTPS = parse_boolean(os.environ.get("REDASH_ENFORCE_HTTPS", "false"))
+ENFORCE_HTTPS_PERMANENT = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_HTTPS_PERMANENT", "false"))
+# Whether file downloads are enforced or not.
+ENFORCE_FILE_SAVE = parse_boolean(
+    os.environ.get("REDASH_ENFORCE_FILE_SAVE", "true"))
+
+COOKIE_SECRET = os.environ.get(
+    "REDASH_COOKIE_SECRET", "c292a0a3aa32397cdb050e233733900f")
+# Whether the session cookie is set to secure.
+SESSION_COOKIE_SECURE = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_SECURE") or str(ENFORCE_HTTPS))
+# Whether the session cookie is set HttpOnly.
+SESSION_COOKIE_HTTPONLY = parse_boolean(
+    os.environ.get("REDASH_SESSION_COOKIE_HTTPONLY", "true"))
+
+# Doesn't set X-Frame-Options by default since it's highly dependent
+# on the specific deployment.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options
+# for more information.
+FRAME_OPTIONS = os.environ.get("REDASH_FRAME_OPTIONS", None)
+FRAME_OPTIONS_ALLOW_FROM = os.environ.get(
+    "REDASH_FRAME_OPTIONS_ALLOW_FROM", "")
+
+# Whether and how to send Strict-Transport-Security response headers.
+# See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
+# for more information.
+HSTS_ENABLED = parse_boolean(os.environ.get("REDASH_HSTS_ENABLED", "false"))
```

Comment:
I'm pretty sure Flask-SSLify was setting it up by default, so maybe the default value should be the one of HTTPS_ENFORCED?

Reply:
That makes sense. For the record, Flask-SSLify only did that when the `request.is_secure` was truthy, which is only the case when the [wsgi environment has `wsgi.url_scheme` set to "https"](https://github.com/pallets/werkzeug/blob/36ed800599f45b74c742b2d9bd3de0ecb93a8547/werkzeug/wrappers/base_request.py#L639-L640), which in turn only happens when `ENFORCE_HTTPS` was `True`.

Reply:
> which in turn only happens when `ENFORCE_HTTPS` was `True`.

Exactly what I meant and suggested. Sorry for not being clear, but when I said it was setting it up by default, I meant when HTTPS is enforced.

Reply:
No worries, I wasn't exactly sure about the behavior, so it's good we spelled it out.